javascript/code-challenges/challenges-05.test.js

```diff
@@ -0,0 +1,377 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1 - Review
+
+Write a function that iterates over an array of people objects
+and creates a new list of each person's full name using the array method 'map'.
+Each object will have the shape {firstName:string, lastName:string}
+E.g. [ { firstName:"Jane", lastName:"Doe" }, { firstName:"James", lastName:"Bond"}]
+should convert to ["Jane Doe", "James Bond"]
+Note the space in between first and last names.
+You can assume that neither firstName nor lastName will be blank
+------------------------------------------------------------------------------------------------ */
+const toLastNames = people => {
+  return people.map(element => {
+    return element.firstName + ' ' + element.lastName;
+  });
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named addValues that, given an array of numbers as input, uses reduce to add the values in the array.
+
+------------------------------------------------------------------------------------------------ */
+
+const addValues = (arr) => {
+  return arr.reduce((a, v) => {
+    return a += v;
+  }, 0);
+};
+
+
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function named addPurchases that, given an array of objects as input, uses reduce to find the total amount purchased. Each object contains the keys `item` and `purchasePrice` like the example.
+
+{
+  item: 'switch'
+  purchasePrice: 399
+}
+
+------------------------------------------------------------------------------------------------ */
+
+const addPurchases = (arr) => {
+  return arr.reduce((a, v) => {
+    return a += v.purchasePrice;
+  }, 0);
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+Write a function named countNumberOfElements that, given an array as input, uses reduce to count the number of elements in the array.
+
+Note: You may not use the array's built-in length property.
+------------------------------------------------------------------------------------------------ */
+
+const countNumberOfElements = (arr) => {
+  return arr.reduce((a, v) => {
+    a = a + v;
+    let total = arr.push();
+    return total;
+  }, 0);
```

Comment:
There's actually a much simple way to do this where reduce does all the heavy lifting for you!
```js
arr.reduce((answer) => answer + 1);
```
Since all you're wanting to do here is give the length, by adding 1 to your return for each type of data, you'll end up returning the correct number.